### PR TITLE
[ESQL] Fix cases of geometry collections with one point in CartesianPoint queries

### LINKED ISSUE #110982: [CI] SpatialPushDownCartesianPointIT testPushedDownQueriesSingleValue failing
**Build Scans:**
- [elasticsearch-periodic-platform-support #3452 / sles-12_platform-support-unix](https://gradle-enterprise.elastic.co/s/4hyr2hzdyynaq)
- [elasticsearch-pull-request #26549 / part-3](https://gradle-enterprise.elastic.co/s/ui3sfnai3i6wa)
- [elasticsearch-pull-request #26214 / part-3](https://gradle-enterprise.elastic.co/s/styyj5sw4qgg2)
- [elasticsearch-periodic-platform-support #3424 / rhel-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/ou5bteqquufka)
- [elasticsearch-periodic-platform-support #3420 / sles-12_platform-support-unix](https://gradle-enterprise.elastic.co/s/cv4fqbew4ftuu)
- [elasticsearch-periodic-platform-support #3416 / ubuntu-2004-aarch64_checkpart3_platform-support-arm](https://gradle-enterprise.elastic.co/s/55opws7gggr4e)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:internalClusterTest" --tests "org.elasticsearch.xpack.esql.spatial.SpatialPushDownCartesianPointIT.testPushedDownQueriesSingleValue" -Dtests.seed=30EFA0EC7D4D6A68 -Dtests.locale=ar-DZ -Dtests.timezone=America/Tijuana -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testPushedDownQueriesSingleValue'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.spatial.SpatialPushDownCartesianPointIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: expected:<0> but was:<5>
```

**Issue Reasons:**
- [main] 6 failures in test testPushedDownQueriesSingleValue (1.0% fail rate in 595 executions)
- [main] 2 failures in step sles-12_platform-support-unix (22.2% fail rate in 9 executions)
- [main] 2 failures in step part-3 (0.8% fail rate in 243 executions)
- [main] 4 failures in pipeline elasticsearch-periodic-platform-support (40.0% fail rate in 10 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (0.8% fail rate in 245 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

### PR DESCRIPTION
**Pull request by iverase**
In cases when contains query geometry is a point represented with  a geometry collection or a multi point, we return a `MatchNoDocsQuery` for cartesian point in ESQL. This is wrong as in fact can match documents. This commit unwraps the collection before doing the check.

fixes https://github.com/elastic/elasticsearch/issues/110982

### CODE REVIEW

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/querydsl/query/SpatialRelatesQuery.java [hunk 1]
```diff
@@ -228,10 +227,10 @@ private static org.apache.lucene.search.Query pointShapeQuery(
             if (geometry == null || geometry.isEmpty()) {
                 throw new QueryShardException(context, "Invalid/empty geometry");
             }
-            if (geometry.type() != ShapeType.POINT && relation == ShapeField.QueryRelation.CONTAINS) {
+            final XYGeometry[] luceneGeometries = LuceneGeometriesUtils.toXYGeometry(geometry, t -> {});
+            if (isPointGeometry(luceneGeometries) == false && relation == ShapeField.QueryRelation.CONTAINS) {
```
Comment (craigtaverner) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/querydsl/query/SpatialRelatesQuery.java:231]: I imagine an edge case where we have a multi-point, but all points are the same, or at least quantize to the same point.

**Review by craigtaverner — APPROVED** [review]
A definite improvement, although I can still imagine an edge case it does not cover. In any case, it's much better than before.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/querydsl/query/SpatialRelatesQuery.java (hunk 1 above)
Comment (iverase) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/querydsl/query/SpatialRelatesQuery.java:231]: indeed, this requires a bit more work but this is still an improvement.

Comment (craigtaverner) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/querydsl/query/SpatialRelatesQuery.java:231]: I see this fix was already in the geo_point case from years ago. So this increases symmetry too.

**Pull request merged**